web/api/v1/api.go

```diff
@@ -692,7 +696,7 @@ func (api *API) labelNames(r *http.Request) apiFuncResult {
 		if len(matcherSets) == 1 {
 			matchers = matcherSets[0]
 		}
-		names, warnings, err = q.LabelNames(r.Context(), matchers...)
+		names, warnings, err = q.LabelNames(r.Context(), hints, matchers...)
```

Comment:
Note that the warnings logic, the `if len(names) >= limit {` block in 709 is broken, https://github.com/prometheus/prometheus/pull/14116 is to fix that.

If that PR is merged before, those new blocks will become unreachable (if the implementations respect the limit, which is expected).

We'll need to think about how to handle that, maybe ask for `limit+1`...
(same applies to labelsValues and series I think).

Reply:
Good catch. I think requesting for limit+1 makes sense to keep the changes minimal.